react-native-aztec/src/AztecView.js

```diff
@@ -91,59 +87,65 @@ class AztecView extends React.Component {
   }
 
   _onFocus = (event) => {
-    if (!this.props.onFocus) {
-      return;
-    }
-
+    this.focused = true;
     const { onFocus } = this.props;
-    onFocus(event);
+
+    if (onFocus) {
+      onFocus(event);
+    }
   }
   
   _onBlur = (event) => {
+    this.focused = false;
     this.selectionEndCaretY = null;
-    TextInputState.blurTextInput(ReactNative.findNodeHandle(this));
+    const { onBlur } = this.props;
 
-    if (!this.props.onBlur) {
-      return;
+    if (onBlur) {
+      onBlur(event);
     }
-
-    const { onBlur } = this.props;
-    onBlur(event);
   }
 
   _onSelectionChange = (event) => {
-    if ( this.props.onSelectionChange ) {
+    if (this.focused && this.props.onSelectionChange) {
       const { selectionStart, selectionEnd, text } = event.nativeEvent;
       const { onSelectionChange } = this.props;
       onSelectionChange( selectionStart, selectionEnd, text, event );
     }
 
-    if ( this.props.onCaretVerticalPositionChange && 
-      this.selectionEndCaretY != event.nativeEvent.selectionEndCaretY ) {
+    if (this.focused && this.props.onCaretVerticalPositionChange && 
+      this.selectionEndCaretY != event.nativeEvent.selectionEndCaretY) {
         const caretY = event.nativeEvent.selectionEndCaretY;
         this.props.onCaretVerticalPositionChange( event.target, caretY, this.selectionEndCaretY );
         this.selectionEndCaretY = caretY;
     }
   }
 
   blur = () => {
-    TextInputState.blurTextInput(ReactNative.findNodeHandle(this));
+    if (this.focused) {
+      if (Platform.OS === "ios") {
+        UIManager.blur(ReactNative.findNodeHandle(this))
+      } else {
+        this.dispatch(UIManager.getViewManagerConfig('AndroidTextInput').Commands.blurTextInput)
+      }
+    }
+    this.focused = false;
   }
 
   focus = () => {
-    TextInputState.focusTextInput(ReactNative.findNodeHandle(this));
+    if (!this.focused) {
+      if (Platform.OS === "ios") {
+        UIManager.focus(ReactNative.findNodeHandle(this))
+      } else {
+        this.dispatch(UIManager.getViewManagerConfig('AndroidTextInput').Commands.focusTextInput)
+      }
+    }
+    this.focused = true;
   }
 
-  isFocused = () => {
-    const focusedField = TextInputState.currentlyFocusedField();
-    return focusedField && ( focusedField === ReactNative.findNodeHandle(this) );
-  }
+  isFocused = () => this.focused
 
   _onPress = (event) => {
-	if ( ! this.isFocused() ) {
-		this.focus(event); // Call to move the focus in RN way (TextInputState)
```

Comment:
I am a bit afraid of removing this code since it is [really old](https://github.com/wordpress-mobile/react-native-aztec/pull/92). I haven't had the chance to test this with Android yet but this seems like a fix for a possible keyboard dismissing issue on Android.

Reply:
I already reverted those changes